sdks/smart-wallet-sdk/src/smartWallet.ts

```diff
@@ -0,0 +1,81 @@
+import { ChainId } from '@uniswap/sdk-core'
+import { encodeFunctionData } from 'viem'
+
+import { abi } from '../abis/MinimalDelegation.json'
+
+import { ModeType, SMART_WALLET_ADDRESSES } from './constants'
+import { Call, MethodParameters, ExecuteOptions, AdvancedCall } from './types'
+import { CallPlanner, ModeEncoder } from './utils'
+
+/**
+ * Main SDK class hashAuthorizationfor interacting with ERC7821-compatible smart wallets
+ */
+export class SmartWallet {
+  /**
+   * Creates method parameters for executing a simple batch of calls through a smart wallet
+   * @dev does not support opData
+   * @param calls Array of calls to encode
+   * @returns Method parameters with calldata and value
+   */
+  public static encodeCalls(calls: Call[], options: ExecuteOptions = {}): MethodParameters {
+    const mode = this.getModeFromOptions(options)
+    if(mode != ModeType.BATCHED_CALL && mode != ModeType.BATCHED_CALL_CAN_REVERT) {
+      throw new Error(`Invalid mode: ${mode}`)
+    }
+    const planner = new CallPlanner(calls)
+    const data = ModeEncoder.encode(mode, planner)
+    const encoded = encodeFunctionData({
```

Comment:
idea for later, but this could probably just be apart of ModeEncoder?

Reply:
yep, I removed the ModeEncoder and just put it into the top level